dali/util/nvml.h

```diff
@@ -122,37 +122,15 @@ inline void SetCPUAffinity(int core = -1) {
 
   int error = pthread_setaffinity_np(pthread_self(), sizeof(requested_set), &requested_set);
   if (error != 0) {
-    DALI_WARN("Setting affinity failed! Error code: " + to_string(error));
+      DALI_WARN("Setting affinity failed! Error code: " + to_string(error));
   }
 }
 
-
 inline void Shutdown() {
   std::lock_guard<std::mutex> lock(Mutex());
   DALI_CALL(wrapNvmlShutdown());
 }
 
-
-inline bool HasHwDecoder(int device_idx) {
-  nvmlDevice_t device;
-  DALI_CALL(wrapNvmlDeviceGetHandleByIndex_v2(device_idx, &device));
```

Comment:
So if `nvmlDeviceGetHandleByIndex_v2` is not available - nullptr if will throw and end process.